### ISSUE TITLE
Improve Status Representation Next to Project

### ISSUE THREAD

**Issue by thisguy-1** [eclipse/codewind#159]
Problem: [Stopped] and [Building] is confusing and difficult to see depending on size of side explorer. Change wording. Do we need both statuses?

- [ ] Change wording - Possibly "Starting"?

**Comment by sishida** [comment]
If "Stopped" and "Building" are part of the same "Starting" process, "Starting" sounds good to me to replace both "Stopped" and "Building."

**Comment by thisguy-1** [comment]
@tetchel What are all the statuses?

**Comment by tetchel** [comment]
### App Statuses:
Running
Starting
Stopping
Stopped

Debugging
Starting - Debug

Disabled
Unknown

### Build Statuses:

Build Succeeded
Building
Build Failed
Build Queued

Unknown

**Comment by tetchel** [comment]
To be clear, these have been the same since Microclimate days, and are shared by the Eclipse plugin

**Comment by deboer-tim** [comment]
@elsony should be included as he has a strong opinion on keeping these statuses independent internally (since different backends and programming languages will go through slightly sets of  states), but he may also be able to help narrow down the likely combinations of states or come up with shorter statuses.

I don't have any quick suggestions, but agree there are some redundant combinations or cases where only one really matters, and others like "[Starting - Debug] [Build Succeeded]" that are just way too long.

On the technical side we probably need to start by having a single 'app status' that defaults to merging the two statuses as it is today (i.e. '[runtime status] [build status]'), then simplify by focussing on what the user cares about in each case.

**Comment by thisguy-1** [comment]
per our conversation with @tetchel @elsony and @jopit, when a user first builds an app, s/he will only see [Building - xyz], while in the Project overview, which has popped up, they can see that the application state is "unknown - waiting for build". Bellow is still a work in progress but wanted to illustrate what this would look like:
<img width="732" alt="Screen Shot 2019-08-23 at 1 51 03 PM" src="https://user-images.githubusercontent.com/52169824/63613061-ce242480-c5ad-11e9-94ba-4fb7c4c85655.png">

**Comment by sishida** [comment]
@thisguy-1 I see the last update for this issue is in August--is this issue still relevant, or should we close it? Please let me know. Thanks!

**Comment by thisguy-1** [comment]
I want to revisit this with @elsony . I was under the impression that this was being done but I'm not sure if it was ever implemented.

**Comment by tetchel** [comment]
we have not made any code changes for this because we are not totally sure what to do - is the requirement "hide the app status until the initial build has completed"?

**Comment by thisguy-1** [comment]
Yes. Since it would say "stopped" before, which is confusing, then "Building-xyz" should be all they see for the first build.

**Comment by thisguy-1** [comment]
In the overview page, for the initial build, it should say "unknown - waiting for build", per @elsony  instructions. Not sure if should just say "waiting for build". Hope Elson can share his thoughts.

**Comment by jopit** [comment]
If we only want to do this for the initial build, I believe it requires a change in PF, right @elsony?

**Comment by elsony** [comment]
Yes, I think the proposal was we'll only show the build status (but not the app status) at the initial build.  The way to achieve that is that we'll set the project app status to unknown and the IDE will use that to decide not showing the application status.  After the initial build is complete and we start the application, we'll update the app status to something else other than unknown and then the IDE will start to show the app status as well.

**Comment by deboer-tim** [comment]
I think that's a great first step, but IMHO we need to find a couple more afterward. e.g. finding cases where X run state and Y build state can be merged into simpler text, or shortening longer descriptions like "Building - Building image container" (I just made that up) to just "Building image".

**Comment by rajivnathan** [comment]
Turbine work to be done in https://github.com/eclipse/codewind/issues/831

**Comment by malincoln** [comment]
@elsony is this still ongoing?  I see the release set to 0.6.0.  If it is not resolved and still ongoing, pls update release.  Thanks

**Comment by malincoln** [comment]
@thisguy-1 @sishida it looks like this is still ongoing.  Please update the release as 0.6.0 has released.  Thanks.

**Comment by thisguy-1** [comment]
@elsony let's discuss this at our next meeting.

**Comment by thisguy-1** [comment]
To do:
Put Build in front of Running Status
Only have build when necessary (temporary "build succeeded")
Put details at the end? [Bulk Detail]
Hover over application see detail of application status
Find out if we can use icons : [X] [Y] [Detail]

**Comment by tetchel** [comment]
To answer the last point, we can use emojis, but not icons in the VS Code tree items :)

[ 🔨 Building] [🟢 Started] ;)

**Comment by jopit** [comment]
Same for IntelliJ, we can use any unicode character (including unicode emojis) but not icons.

**Comment by thisguy-1** [comment]
We should discuss what are some possible emoji's. We already a use a hammer icon for the command button for build in the codewind view on VSCode and eclipse.
